core/common/src/ChangesetProps.ts

```diff
@@ -53,6 +53,8 @@ export enum ChangesetType {
   Regular = 0,
   /** changeset *does* contain schema changes. */
   Schema = 1,
+  /** Schema changeset pushed by iModel with SchemaSync enabled */
+  SchemaSync = Schema | 64,
```

Comment:
On old code we have condition of equality and not ORing the flag. This mean 65 changeset is detected as Data/Regular changeset if condition say `type===ChangesetType.Schema`. The older code can be on imodel02 and we do not have to worry about 3.x